public/js/actions/GraphActions.js

```diff
@@ -0,0 +1,37 @@
+import dispatcher from '../dispatcher'
+import ActionTypes from '../constants/ActionTypes'
+
+export default class GraphActions {
+  static graphArticle(id) {
+    fetch(`articles/${id}`)
+            .then(res => res.json())
+            .then(article => article.keywords[0])
+            .then(word => this.updateGraph(word))
+  }
+
+  static graphTop() {
+    this.updateGraph('top')
+  }
+
+  static updateGraph(word) {
+    return fetch(`keywords/${word}`)
+            .then(res => res.json())
+            .then((keyword) => {
+              const dataPoints = keyword[0].votes.map(
+                    vote => [vote.time, vote.sum],
+                )
+              dataPoints.unshift(['Time', keyword[0].word])
+              return dataPoints
```

Comment:
You're always returning an synchronous value, there's no need to chain it as a promise